api/src/main/resources/client_api.yaml

```diff
@@ -165,6 +210,139 @@ definitions:
+      cdrVersionName:
+        description: >
+          The name of a CDR version to use when evaluating the cohort; if none is specified,
+          the CDR version currently associated with the workspace will be used
+        type: string
+      tableQuery:
+        description: >
+            A query specifying how to pull data out of a single table. If tableQuery is not
+            specified, just Person.person_id will be extracted.
+        $ref: "#/definitions/TableQuery"
+
+  CdrQuery:
+    type: object
+    required:
+      - sql
+      - bigqueryProject
+      - bigqueryDataset
+
+    properties:
+      sql:
+        description: Google SQL to use when querying the CDR
+        type: string
+      parameters:
+        description: named query parameters used when executing the above SQL
+        type: array
+        items:
+          $ref: "#/definitions/CdrQueryParameter"
+      bigqueryProject:
+        description: name of the Google Cloud project containing the CDR dataset
+        type: string
+      bigqueryDataset:
+        description: name of the CDR BigQuery dataset
+        type: string
+
+  CdrQueryParameter:
+    type: object
+    required:
+      - name
+    description: A query parameter and value. One (and only one) of the value fields should be set.
+    properties:
+      name:
+        description: the name of the parameter
+        type: string
+      value:
+        description: A string value for this parameter.
+        type: string
+      values:
+        description: An array of string values for this parameter.
+        type: array
+        items:
+          type: string
+      valueDate:
+        description: A date (yyyy-MM-dd) value for this parameter
+        type: string
+      valueDatetime:
+        description: A datetime (yyyy-MM-dd HH:mm:ss zzz) value for this parameter
+        type: string
+      valueNumber:
+        description: A number value for this parameter.
+        type: number
+      valueNumbers:
+        description: An array of number values for this parameter
+        type: array
+        items:
+          type: number
+
+  CohortAnnotationsRequest:
+    type: object
+    required:
+      - cohortName
+    properties:
+      cohortName:
+        description: >
+         The name of a cohort that annotations should be retrieved for.
+        type: string
+      statusFilter:
+        description: >
+          An array of status values; participants with these statuses will be have their annotations
+          retrieved. Defaults to [NOT_REVIEWED, INCLUDED, NEEDS_FURTHER_REVIEW] --
+          everything but EXCLUDED.
+        type: array
+        items:
+          $ref: "#/definitions/CohortStatus"
+      cdrVersionName:
+        description: >
+          The name of a CDR version to use when retrieving annotations; if none is specified,
+          the CDR version currently associated with the workspace will be used
+        type: string
+      annotationQuery:
+        description: >
```

Comment:
Would it make sense to make statusFilter above become part of the annotationQuery? Seems like these would be used together in implementation?

Reply:
So the current code has a separation between TableQuery / AnnotationQuery (which indicate what data to get about a person) and MaterializeCohortRequest / DataTableSpecification / CohortAnnotationRequest (which indicate what *people* to get data for, as well as including what data to get about them.)

I like this separation, since you can totally imagine re-using TableQuery or AnnotationQuery in other contexts in future (e.g. "get me the data for just person X"). So I think I want to keep statusFilter separate here.

(Also FWIW: we need to make sure we maintain backwards compatibility for materialize cohort logic until we get rid of it.)